### PR TITLE
Added `addPragma` and `unpackedName` routines.

### PR DESCRIPTION
**Pull request by yglukhov**
Some procs to make metaprogramming easier.

### CODE REVIEW

File: lib/core/macros.nim [hunk 1]
```diff
@@ -667,8 +667,18 @@ template expectRoutine(node: NimNode): stmt =
   expectKind(node, RoutineNodes)
 
 proc name*(someProc: NimNode): NimNode {.compileTime.} =
+  ## Returns name node of the proc. Note: if result is nnkPostfix
+  ## the name is not unpacked. Use `unpackedName` to obtain the "pure" name.
   someProc.expectRoutine
   result = someProc[0]
+
+proc unpackedName*(someProc: NimNode): NimNode {.compileTime.} =
```
Comment (Araq) [inline lib/core/macros.nim]: Pretty sure that's already covered by some other proc in macros.nim.

File: lib/core/macros.nim [hunk 2]
```diff
@@ -692,6 +702,14 @@ proc `pragma=`*(someProc: NimNode; val: NimNode){.compileTime.}=
   assert val.kind in {nnkEmpty, nnkPragma}
   someProc[4] = val
 
+proc addPragma*(someProc, pragma: NimNode) {.compileTime.} =
```
Comment (Araq) [inline lib/core/macros.nim]: This one seems useful and should later be extended to vars, types etc.

File: lib/core/macros.nim (hunk 1 above)
Comment (yglukhov) [inline lib/core/macros.nim]: You probably mean `unpackPostfix`, but that one expects only `nnkPostfix` nodes.

**Comment by yglukhov** [comment]
`unpackedName` removed. New pr with just `addPragma`: #4394

**Pull request closed without merging**